File: cs3/ncr.cc
#include<stdio.h>
#include<math.h>
#include <iostream>
using namespace std;
int main()
{
    unsigned long long int t,i,n,r,c,num = 1,den = 1,temp;
    cin>>t;
    for(i=0;i<t;i++)
    {
        cin>>n;
        cin>>r;
        if(r > n) cout<<"\n0";
        if(n == 0) cout<<"\n0";
        c = n-r;
        if(c > r)
        {
            temp = r;
            r = c;
            c = temp;
        }
        for(i=r+1;i<=n; i++)
        {
            num = (num * i)%142857;
        }
        cout<<num<<"\n";
        for(i=2;i<=c;i++)
        {
            den = (den * i)%142857;
        }
        cout<<den;
        cout<<"\n";
        cout<<num/den;
    }
    return 0;
}

File: cs3/ncr.py
from math import factorial,floor
t = input();

for i in range (0, t):
    s = raw_input().split();
    n = long(s[0])
    r = long(s[1])
    c = n-r
    if(c>r):
        temp = r;
        r = c;
        c = temp
    num = den = 1
    for j in range(r+1, n+1):
        num = (num * j)
    for j in range(2, c+1):
        den = (den * j)
    print (num/den)%142857